### PR TITLE
Create new vsix for x64 .NET ServiceHub services

### PR DESCRIPTION
**Pull request by genlu**
prerequisite of #64610
link to validation insertion (with additional setup authoring change on VS side)
https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/468984

### CODE REVIEW

File: Roslyn.sln [hunk 1]
```diff
@@ -517,12 +515,16 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CommonLanguageSer
...
+Project("{778DAE3C-4631-46EA-AA77-85C1314464D9}") = "Microsoft.CodeAnalysis.VisualBasic.LanguageServer.Protocol", "src\Features\LanguageServer\Microsoft.CodeAnalysis.VisualBasic.LanguageServer.Protocol\Microsoft.CodeAnalysis.VisualBasic.LanguageServer.Protocol.vbproj", "{F836BDB1-4EC7-4F4C-B2E9-BCD721C9E650}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TestDiscoveryWorker", "src\Tools\TestDiscoveryWorker\TestDiscoveryWorker.csproj", "{8BC50AFF-1EBF-4E9A-AEBB-04F387AA800F}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Roslyn.VisualStudio.ServiceHub.Setup.x64", "src\VisualStudio\Setup.ServiceHub\x64\Roslyn.VisualStudio.ServiceHub.Setup.x64.csproj", "{A900A17C-3FD1-4892-8528-51A967E77339}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.x64", "src\Workspaces\Remote\ServiceHub.CoreComponents\x64\Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.x64.csproj", "{3ABBDFF7-ED76-4EA3-840E-85E3274302B0}"
+EndProject
```
Comment (jasonmalinowski) [inline Roslyn.sln:527]: Why do we need two projects here?

Comment (genlu) [reply Roslyn.sln:527]: "Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.x64" is used to collect all dependencies and crossgen them. "Roslyn.VisualStudio.ServiceHub.Setup.x64" is to create the vsix. Maybe there's a better way to do it w/o multiple projects, but I have no idea how :(

File: src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
         - compile ReadyToRun binaries
-    -->
-    <PackageReference Include="Microsoft.NETCore.App.Runtime.win-x64" Version="$(MicrosoftNETCoreAppRuntimewinx64Version)" ExcludeAssets="all" PrivateAssets="all" GeneratePathProperty="true" />
-    <PackageReference Include="Microsoft.WindowsDesktop.App.Runtime.win-x64" Version="$(MicrosoftWindowsDesktopAppRuntimewinx64Version)" ExcludeAssets="all" PrivateAssets="all" GeneratePathProperty="true" />
-    <PackageReference Include="Microsoft.NETCore.App.crossgen2.win-x64" Version="$(MicrosoftNETCoreAppcrossgen2winx64Version)" Condition="'$(GenerateReadyToRun)' == 'true'" ExcludeAssets="all" PrivateAssets="all" GeneratePathProperty="true" />
+
+      "Microsoft.NETCore.App.Runtime.win-{arch}"
+      "Microsoft.WindowsDesktop.App.Runtime.win-{arch}"
+    -->    
+    <PackageReference Include="Microsoft.NETCore.App.crossgen2.win-x64" Version="$(MicrosoftNETCoreAppcrossgen2winx64Version)" Condition="'$(GenerateReadyToRun)' == 'true'" ExcludeAssets="all" PrivateAssets="all" GeneratePathProperty="true" />
```
Comment (tmat) [inline src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets:32]: >win-x64 [](http://example.com/codeflow?start=63&length=7)

Shouldn't this use TargetArch?

Comment (genlu) [reply src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets:32]: yea, you are right. I will fix it in the follow up when adding arm64 support. Thanks!

**Pull request merged**